testing/dart/canvas_test.dart

```diff
@@ -198,4 +217,22 @@ void main() {
         await fuzzyGoldenImageCompare(image, 'canvas_test_dithered_gradient.png');
     expect(areEqual, true);
   }, skip: !Platform.isLinux); // https://github.com/flutter/flutter/issues/53784
+
+  test('Image size reflected in picture size for image* and drawPicture methods', () async {
+    final Image image = await createImage(100, 100);
+    final PictureRecorder recorder = PictureRecorder();
+    final Canvas canvas = Canvas(recorder);
+    const Rect rect = Rect.fromLTWH(0, 0, 100, 100);
+    canvas.drawImage(image, Offset.zero, Paint());
+    canvas.drawImageRect(image, rect, rect, Paint());
+    canvas.drawImageNine(image, rect, rect, Paint());
+    final Picture picture = recorder.endRecording();
+    expect(picture.approximateBytesUsed, 161235);
```

Comment:
I'd break down the calculation of these numbers.

Reply:
That'll be more possible if we expose this on Canvas, which as I think about it, we should probably do.  @chinmaygarde does that SGTY? While the Canvas is still alive, it holds a reference to the SkImage in this case.

Reply:
We don't really expose as much on the Dart side here either - for example, we don't expose the image allocation size

Reply:
(and if there's something I'm missing here that you think would just make the test better I'm open to suggestions)

Reply:
I meant like 
```dart
int expectedBytes = /*foo*/ 123 + /*bar*/ 567 + (4 * /*baz*/ 89);  
expect (picture.approximateBytesUsed, expectedBytes);
```

Reply:
Ahh ok, I can do that :)